packages/uui-modal/lib/uui-modal.element.ts

```diff
@@ -0,0 +1,114 @@
+import { LitElement, css } from 'lit';
+import { customElement, property, query } from 'lit/decorators.js';
+
+@customElement('uui-modal')
+export class UUIModalElement extends LitElement {
+  @query('dialog')
+  protected _dialogElement?: HTMLDialogElement;
+
+  @property({ type: Boolean, reflect: true, attribute: 'is-open' })
+  isOpen = false;
+
+  @property({ type: Boolean, reflect: true, attribute: 'is-closing' })
+  isClosing = false;
+
+  @property({ type: Number, reflect: true })
+  index = 0;
+
+  @property({ type: Number, reflect: true, attribute: 'unique-index' })
+  uniqueIndex = 0;
+
+  private _transitionDuration = 250;
+
+  @property()
+  public get transitionDuration() {
+    return this._transitionDuration;
+  }
+  public set transitionDuration(value) {
+    this._transitionDuration = value;
+    this.style.setProperty(
+      '--uui-modal-transition-duration',
+      this._transitionDuration + 'ms'
+    );
+  }
+
+  protected firstUpdated(
+    _changedProperties: Map<string | number | symbol, unknown>
+  ): void {
+    super.firstUpdated(_changedProperties);
+
+    this.open();
```

Comment:
Could it make sense to make a check for isClosing there, so we don't open if its already closing?